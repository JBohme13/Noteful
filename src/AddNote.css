#form-container {
    display: flex;
    flex-direction: column;
}

#type-note-here {
    white-space: pre-wrap;
    height: 300px;
    width: 80%;
    flex: 1;
    order: 3;
    font-size: 14px;
    margin: 5px;
}

#note-name {
    order: 1;
    flex: 1;
    height: 30px;
    width: 55%;
    color: darkslategrey;
    margin: 5px;
}

#select-folder {
    order: 2;
    flex: 2;
    height: 30px;
    width: 55%;
    margin: 5px;
    font-size: 14px;
    color: darkslategrey;
}

#note-submit {
    order: 4;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#note-submit:active {
    box-shadow: -1px -1px;
}

#note-clear {
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#note-clear:active {
    box-shadow: -1px -1px;
}

label {
    color: darkslategrey;
}

h2 {
    color: darkslategrey;
    font-weight: normal;
}

