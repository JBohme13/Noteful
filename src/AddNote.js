import React, { Component } from 'react'
import NotefulContext from './NotefulContext'
import ValidationError from './ValidationError'
import PropTypes from 'prop-types'
import './AddNote.css'

export default class AddNote extends Component {
    static contextType = NotefulContext;

    validateNameInput = () => {
      const name = this.context.noteName.value.trim();
      if ( name.length === 0) {
        return 'Name is required'
      } else if (name.length <= 2) {
        return 'Note name must be at least three characters'
      }
    };
  
    validateNoteInput = () => {
      const body = this.context.noteBody.value.trim();
      if (body.length === 0) {
        return 'Enter valid message'
      } else if ( body.length > 3000) {
        return 'Maximum characters is 3000'
      }
    };

    render() {
        const value = this.context;
        const nameError = this.validateNameInput();
        const bodyError = this.validateNoteInput();
        return(
          <form id='form-container'>
              <h2>Add new note</h2>
              <section id='form-group'>
                  <label htmlFor='note-name'>Note name</label><br/>
                  <input 
                    type='text'
                    name='note-name'
                    id='note-name'
                    required='required'
                    aria-required='true'
                    onChange={event => value.handleNameChange(event.target.value)}
                  />
                  {value.noteName.touched && <ValidationError message={nameError} />}
                <br/>
                  <label htmlFor='select-folder'>Select a folder</label><br/>
                  <select 
                    type='radio'
                    name='select-folder'
                    id='select-folder'
                    required='required'
                    aria-required='true'
                  >
                      {
                          value.folders.map((folder, i) => {
                            return(
                              <option 
                                key={i}
                                value={folder.id}
                              >
                                  {folder.name}
                              </option>
                            )
                          })
                      }
                  </select><br/>
                  <label htmlFor='type-note-here'>Type note here</label><br/>
                  <textarea 
                    id='type-note-here'
                    onChange={event => value.handleNoteChange(event.target.value)}
                  >
                  </textarea>
                  {value.noteBody.touched && <ValidationError message={bodyError} />}
                  <br/>
                  <button 
                    id='note-submit'
                    type='submit'
                    disabled = {
                      nameError ||
                      bodyError
                    }
                    onClick={event => {
                        return value.addNote(event, document.getElementById('select-folder').value)}
                    }
                  >
                      Submit
                  </button>
                  <button 
                    id='note-clear'
                    type='reset'
                    onClick={event => value.handleClearButton(event)}
                    >
                      Clear
                    </button>
              </section>
          </form>
        )
    }
};
AddNote.propTypes = {
  validateNameInput: PropTypes.func,
  validateNoteInput: PropTypes.func,
  handleNameChange: PropTypes.func,
  handleNoteChange: PropTypes.func,
  handleClearButton: PropTypes.func,
  addNote: PropTypes.func,
  folders: PropTypes.array,
  noteName: PropTypes.object,
  noteBody: PropTypes.object,
};

AddNote.defaultProps = {
  folders: [],
  noteName: {},
  noteBody: {},
  addNote: () => {},
  handleClearButton: () => {},

}