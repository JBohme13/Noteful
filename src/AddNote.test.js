import React from 'react';
import ReactDOM from 'react-dom';
import AddNote from './AddNote';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

describe('AddNote component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddNote />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AddNote />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  /*it('adds a new note', () => {
    const wrapper = shallow(<AddNote/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })*/
});