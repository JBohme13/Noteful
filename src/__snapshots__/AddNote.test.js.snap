// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component renders the UI as expected 1`] = `
<form
  id="form-container"
>
  <h2>
    Add new note
  </h2>
  <div
    id="form-group"
  >
    <label
      htmlFor="note-name"
    >
      Note name
    </label>
    <br />
    <input
      id="note-name"
      name="note-name"
      required="required"
      type="text"
    />
    <br />
    <label
      htmlFor="select-folder"
    >
      Select a folder
    </label>
    <br />
    <select
      id="select-folder"
      name="select-folder"
      required="required"
      type="radio"
    />
    <br />
    <label
      htmlFor="note-body"
    >
      Type note here
    </label>
    <br />
    <textarea
      id="note-body"
    />
    <br />
    <button
      id="note-submit"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
    <button
      id="note-clear"
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
</form>
`;
